src/main/java/com/reproio/embulk/input/trino/TrinoJdbcInputPlugin.java

```diff
@@ -0,0 +1,83 @@
+package com.reproio.embulk.input.trino;
+
+import io.trino.jdbc.TrinoDriver;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Optional;
+import java.util.Properties;
+import org.embulk.input.jdbc.AbstractJdbcInputPlugin;
+import org.embulk.input.jdbc.JdbcInputConnection;
+import org.embulk.util.config.Config;
+import org.embulk.util.config.ConfigDefault;
+
+public class TrinoJdbcInputPlugin extends AbstractJdbcInputPlugin {
```

Comment:
If only this class can work, it may be possible to contribute this feature to upstream (embulk-input-jdbc).